src/pages/index.js

```diff
@@ -1,55 +1,160 @@
 import "./index.css";
 import Card from "../components/Card.js";
-import { cardList, config, userInfo } from "../utils/constants.js";
+import { config, userInfo } from "../utils/constants.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
 
-// What is a computer virus?
-// A terminal illness.
+let currentCard = undefined;
 
-// Popups
-const imagePopup = new PopupWithImage({
-  popupSelector: ".popup--open-image",
-});
-const userInfoPopup = new PopupWithForm(
-  { popupSelector: ".popup--edit-profile" },
-  (data) => userInfo.setUserInfo({ name: data.name, job: data.job })
-);
-const newPlacePopupButton = document.querySelector(".profile__add-button");
-const editInfoPopupButton = document.querySelector(
-  ".profile__edit-name-button"
+// Delete card Popup
+const deleteCardPopup = new PopupWithForm(
+  { popupSelector: ".popup--delete-card" },
+  () => {
+    deleteCardPopup.getForm().querySelector(".form__submit").textContent =
+      "Deleting...";
+    return api
+      .deleteCard({
+        cardId: currentCard.getCard().id,
+      })
+      .then(() => {
+        renderCards();
+        deleteCardPopup.close();
+      });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**